### PR TITLE
optimize answer responses containing long choice lists

### PR DESCRIPTION
**Pull request by dannyroberts**
by not updating the choices if they haven't changed

### CODE REVIEW

**Comment by ctsims** [comment]
@dannyroberts you rule, thanks for getting this out so fast! 

not sure what the knockout equality op does on these sets, the list order should be guaranteed from the DB, since this is in in XML land at this point, but not sure if he widget ordering retains that order specifically since it's an autocomplete/combobox.

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/fullform-ui.js [hunk 1]
```diff
@@ -162,13 +162,19 @@ Container.prototype.fromJS = function(json) {
                 if (ko.utils.unwrapObservable(options.target.serverError)) {
                     options.data.answer = _.clone(options.target.answer());
                 }
+                if (_.isEqual(options.target.choices(), options.data.choices)) {
```
Comment (orangejenny) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/fullform-ui.js]: Is there anything like an id or path in the array elements that you could pluck out to make the comparison quicker?

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/fullform-ui.js (hunk 1 above)
Comment (dannyroberts) [reply corehq/apps/cloudcare/static/cloudcare/js/form_entry/fullform-ui.js]: Ha, so actually choices is an array of strings, not of the richer objects you might expect, so there's nothing to compare on but these string values themselves.

**Comment by dannyroberts** [comment]
@ctsims, yeah this `_.isEqual` is just a utility (part of the underscore.js or lodash.js library) that is there to make up for the fact that javascript equality is rarely ever what you want it to be, specifically  stuff like `_.isEqual([1], [1])` returns `true` where `[1] == [1]` returns `false` (because `==`/`===` in this case uses object equality). Anyway,

```js
> _.isEqual([1, 2], [1, 2]);
true
> _.isEqual([1, 2], [2, 1]);
false
```
so there's nothing to worry about—in the edge case where the order changes, the dropdown will go through the (slow) update process.

**Review by gcapalbo — APPROVED** [review]

**Pull request merged**